ionoscloud/resource_server.go

```diff
@@ -45,6 +45,12 @@ func resourceServer() *schema.Resource {
 				Required:         true,
 				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
 			},
+			"hostname": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Description:      "The hostname of the resource. Allowed characters are a-z, 0-9 and - (minus). Hostname should not start with minus and should not be longer than 63 characters.",
+				ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.StringIsNotWhiteSpace, validation.StringLenBetween(1, 63))),
```

Comment:
@cristiGuranIonos same discussion about validation. Here we check the length of the string, we are not checking the content of the string, basically we do a partial check. My suggestion: remove this `ValidateDiagFunc` and let the API do all the validation.

Reply:
will leave the partial check, doesn't hurt anything  even if the validation changes for this.